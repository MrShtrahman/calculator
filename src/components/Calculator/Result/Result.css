.result {
    padding-left: 10px;
    font-size: x-large;
    font-weight: bold;
}