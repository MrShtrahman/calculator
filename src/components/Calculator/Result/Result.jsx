import { Form, FormInput } from 'semantic-ui-react';

import React from 'react';
import { useSelector } from 'react-redux'

const Result = () => {
  const display = useSelector(state => state.calculatorBasicReducer.display)
    return (
      <Form>
        <FormInput fluid readOnly value = {display}/>
      </Form>
    )
}

export default Result;
