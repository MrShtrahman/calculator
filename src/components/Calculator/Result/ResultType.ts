export type ResultType = {
  display: string;
  isDarkMode: boolean
};
