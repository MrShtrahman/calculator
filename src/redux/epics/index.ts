import { combineEpics } from "redux-observable";
import historyEpic from "./HistoryEpic";

const epics = combineEpics(...historyEpic);

export default epics;
