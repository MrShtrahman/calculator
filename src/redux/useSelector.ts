import {
  TypedUseSelectorHook,
  useSelector as useReduxSelector
} from "react-redux";

import reducer from "./reducers/rootReducer";

type RootState = ReturnType<typeof reducer>;

export const useSelector: TypedUseSelectorHook<RootState> = useReduxSelector;
