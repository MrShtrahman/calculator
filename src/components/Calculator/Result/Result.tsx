import { Form, FormInput } from "semantic-ui-react";
import React, { FC } from "react";

import { ResultType } from "./ResultType";

const Result: FC<ResultType> = ({ display }) => {
  return (
    <Form>
      <FormInput fluid readOnly data-testid="result-label">
        {display}
      </FormInput>
    </Form>
  );
};

export default Result;
