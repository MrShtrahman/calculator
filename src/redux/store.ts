import { applyMiddleware, compose, createStore } from "redux";
import rootReducer, { RootAction, RootState } from "./reducers/rootReducer";

import { createEpicMiddleware } from "redux-observable";
import epics from "./epics";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: Function;
  }
}
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const epicMiddleware = createEpicMiddleware<
  RootAction,
  RootAction,
  RootState
>();

const configureStore = (initialState?: RootState) => {
  const middlewares = [epicMiddleware];
  const enhancer = composeEnhancers(applyMiddleware(...middlewares));
  return createStore(rootReducer, initialState, enhancer);
};

const store = configureStore();
epicMiddleware.run(epics);

export default store;
