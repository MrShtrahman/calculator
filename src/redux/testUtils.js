import { configure, mount } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import React from "react";
import { createStore } from "redux";
import reducer from "./reducers/rootReducer";
import { render } from "@testing-library/react";

configure({
  adapter: new Adapter()
});

const initialState = {
  memo: {
    memo: 0,
    isMemoShown: false
  },
  calcBasic: {
    left: 0,
    right: 0,
    display: "0",
    operator: ""
  },
  calcMetadata: {
    isLeft: true,
    equalClicked: false,
    operatorClicked: false,
    history: ""
  }
};

const store = createStore(reducer, initialState);

export const renderWithRedux = ui => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store
});

export const mountWithRedux = ui =>
  mount(<Provider store={store}>{ui}</Provider>);
