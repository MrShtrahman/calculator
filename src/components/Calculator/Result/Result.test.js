import React from "react";
import Result from "./Result";
import { mountWithRedux } from "../../../redux/testUtils";

describe("<Result /> component", () => {
  it("renders", () => {
    const wrapper = mountWithRedux(<Result />);

    expect(wrapper).toMatchSnapshot();
  });
});
