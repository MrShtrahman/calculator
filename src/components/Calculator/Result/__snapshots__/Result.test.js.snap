// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> component renders 1`] = `ReactWrapper {}`;
